tests/unit/tf/models/test_retrieval.py

```diff
@@ -18,6 +20,64 @@
 from tests.common.tf.retrieval import retrieval_tests_common
 
 
+def test_two_tower_shared_embeddings():
+    train = make_df(
+        {
+            "user_id": [1, 3, 3, 4, 3, 1, 2, 4, 6, 7, 8, 9] * 100,
+            "item_id": [1, 2, 3, 4, 11, 12, 5, 1, 1, 3, 5, 11] * 100,
+            "item_id_hist": [
+                [1, 3, 10],
+                [1, 5],
+                [4, 2, 1],
+                [1, 2, 3],
+                [1],
+                [3, 4],
+                [1, 3, 10],
+                [11, 3, 10],
+                [3, 4],
+                [1, 3, 10],
+                [11, 3, 10],
+                [1, 11],
+            ]
+            * 100,
+        }
+    )
+
+    user_id = ["user_id"] >> nvt.ops.Categorify() >> nvt.ops.TagAsUserID()
+
+    joint_feats = [["item_id_hist", "item_id"]] >> nvt.ops.Categorify()
+
+    item_id = joint_feats["item_id"] >> nvt.ops.TagAsItemID()
+    user_feat = joint_feats["item_id_hist"] >> nvt.ops.TagAsUserFeatures()
+    outputs = user_id + item_id + user_feat
+
+    train_dataset = Dataset(train)
+
+    workflow = nvt.Workflow(outputs)
+    workflow.fit(train_dataset)
+    train = workflow.transform(train_dataset)
+    schema = train.schema
+
+    embeddings = mm.Embeddings(schema)
```

Comment:
Wouldn't it be easier to create a single `InputBlockV2` and use `select_by_tag` on that to get the `user_inputs` & `item_inputs`?

Reply:
I've changed this to use the inputblock select_by_tag and it turns out it didn't work. Fixes to make this work in #851